packages/istanbul-reports/lib/html/annotator.js

```diff
@@ -57,13 +57,12 @@ function annotateStatements(fileCoverage, structuredText) {
 
         if (type === 'no' && structuredText[startLine]) {
             if (endLine !== startLine) {
-                endLine = startLine;
                 endCol = structuredText[startLine].text.originalLength();
             }
             text = structuredText[startLine].text;
             text.wrap(startCol,
                 openSpan,
-                startLine === endLine ? endCol : text.originalLength(),
+                startCol < endCol ? endCol : text.originalLength(),
```

Comment:
Yup, this means the existing logic is intact and only fixes the wrong scenario. Although still not ideal it means that if the source maps side of things gets fixed no change is required here.